subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/TestNGExecutionResult.groovy

```diff
@@ -144,10 +144,20 @@ class TestNgTestClassExecutionResult implements TestClassExecutionResult {
         this
     }
 
+    @Override
+    TestClassExecutionResult assertTestFailed(String name, String displayName, Matcher<? super String>... messageMatchers) {
```

Comment:
⭕️ Instead of adding more overloads with additional parameters, how about we introduce a `Matcher<TestCase>` here?

Reply:
I'd rather add overloads. For HTML reports, a test case's name and display name is extracted from `<h3>` tag and stdout/stderr is extracted from `<pre>` tag. `messageMatchers` aims at matching message lines, they overlap little and I think they should be separated.

Reply:
I'm suggesting a data object that contains name, display name and message. Then this method would take a single Matcher that can inspect all that data. That way you can match the simple name, display name and message any way you like. I'll leave it up to you. If all tests just want to assert the exact name then the overload approach may be fine.